app/model/DigitalEdition.scala

```diff
@@ -0,0 +1,25 @@
+package model
+
+import configuration.Config
+
+case class DigitalEdition(id: String, name: String, price: String, campaign: String)
+
+
+object DigitalEdition {
+
+  trait UrlConfig {
+    val externalSubscriptionUrl: String
+  }
+
+  object UK extends DigitalEdition("uk", "UK", "£11.99", "dis_2408")
+  object US extends DigitalEdition("us", "US", "$19.99", "dis_2378")
+  object AU extends DigitalEdition("au", "Australia", "$21.50", "dis_2379")
+  implicit lazy val defaultUrlConfig: UrlConfig = Config
+
+  def getRedirect(edition: DigitalEdition)(implicit urlConfig: UrlConfig): String = {
```

Comment:
Ah that ```java.net.URL``` tip is a good one!

With the use of the ```UrlConfig``` trait - if the urlConfig argument becomes a baseUrl string then the controller calling getRedirect will have to become dependant on the global Config super object to retrieve the correct value to use.

By using a trait like this if we can move the dependency on Config from the caller to

```scala
implicit lazy val defaultUrlConfig: UrlConfig = Config
```

Which keeps the controller free of global state and allows us to swap out the source of Config in one place if the function ever gets repeatedly called from different scopes

The use of an ```implicit val``` to inject the chosen implementation of ```UrlConfig``` is pretty dubious on my part, you'd achieve the same thing with just a normal val somewhere.

I know this is a lot of complexity for a very simple feature, I did it like this to find out what people thought of dependency injection patterns such as this in general.

Given that the controllers are unlikely ever to be unit tested I'll swap to using a ``java.net.URL``` URL argument for simplicity's sake :smile: